lib/MCToIRLifter.cpp

```diff
@@ -217,6 +246,29 @@ void MCToIRLifter::VisitDirectFunctionCall(const remill::Instruction &inst,
   VisitAfterFunctionCall(inst, block);
 }
 
+void MCToIRLifter::VisitConditionalDirectFunctionCall(const remill::Instruction &inst,
+                               remill::Instruction *delayed_inst,
+                               llvm::BasicBlock *block) {
+  const auto lifted_func = block->getParent();
+  VisitDelayedInstruction(inst, delayed_inst, block, true);
```

Comment:
The `true` parameter here means: "lift the delayed instruction assuming the branch is taken", but I think you want to do both `true` and `false` delayed instruction handling here, as you had done with the indirect conditional branch[

Reply:
```
case Instruction::kCategoryConditionalDirectFunctionCall: {
          if (inst.branch_not_taken_pc == inst.branch_taken_pc) {
            goto direct_func_call;
          }
          try_add_delay_slot(true, block);
          trace_work_list.insert(inst.branch_taken_pc);
          auto target_trace = get_trace_decl(inst.branch_taken_pc);
          auto do_cond_call = llvm::BasicBlock::Create(context, "", func);
          auto next_block = GetOrCreateBranchNotTakenBlock();
          llvm::BranchInst::Create(do_cond_call, next_block,
                                   LoadBranchTaken(block), block);
          AddCall(do_cond_call, target_trace);

          const auto ret_pc_ref = LoadReturnProgramCounterRef(block);
          const auto next_pc_ref = LoadNextProgramCounterRef(block);
          llvm::IRBuilder<> ir(do_cond_call);
          ir.CreateStore(ir.CreateLoad(ret_pc_ref), next_pc_ref);
          ir.CreateBr(next_block);

          continue;
        }
```
I only add true based on there being only `try_add_delay_slot(true, block);` in what was added in the remill TraceLifter. Should there also be a `try_add_delay_slot` for `false` underneath the `try_add_delay_slot` for `true`?

Reply:
There should be a `false` case becasue these are conditional calls; in an unconditional case, the call is always taken. The thing with delay slots on conditional control flow is that they have different "anulment" behaviour when the flow is taken vs. not-taken.

Reply:
But yes, the tracelifter is probably also wrong.